tools/lib/esbuildConfig.js

```diff
@@ -1,58 +1,53 @@
 import fs from "fs/promises";
 import babel from "@babel/core";
 import babelPluginSyntaxTypescript from "@babel/plugin-syntax-typescript";
-import babelPluginSportFunctions from "../babel-plugin-sport-functions/index.cjs";
-import { getSport } from "./buildFuncs.js";
+import babelPluginSportFunctions from "../babel-plugin-sport-functions/index.js";
+import getSport from "./getSport.js";
+import path from "path";
 
 const babelCache = {};
 
 // Use babel to run babel-plugin-sport-functions. This is needed because the way bySport is defined, the sport-specific code will run if it's present, which can produce errors. It's not actually needed for isSport.
 const pluginSportFunctions = nodeEnv => ({
 	name: "sport-functions",
 	setup(build) {
-		build.onLoad({ filter: /\.tsx?$/, namespace: "file" }, async args => {
+		build.onResolve({ filter: /\..{1,}Sport/ }, async args => {
```

Comment:
The filter runs on the filename, not on the contents of the file, so I think this will never run because no filename matches that regex.

Reply:
I believe the on resolve callback will be called when we try to 'resolve' the import. So when an import is found in a file, esbuild will try and inline the code from the imported file using our defined callbacks.

---

This filter will invoke the callback when we find an import with the `path` field, in the [esbuild arguments](https://esbuild.github.io/plugins/#on-resolve-arguments), equal to .\**Sport. This should handle any import for isSport and bySport, I think.

---

There might be cases where the path field is not .\isSport or .\bySport. Might be cases where path is ..\bySport, bySport, or something similar that could be missed.

Reply:
what do you think this filter should be?

Reply:
It's unclear to me what the filter is trying to accomplish. But ultimately, if it works, then I'll figure out why. So make a PR from the latest master so it actually runs and then we'll see :)